common/src/main/java/com/wynntils/wynn/model/quests/QuestInfo.java

```diff
@@ -0,0 +1,177 @@
+ */
+package com.wynntils.wynn.model.quests;
+
+import com.wynntils.mc.objects.Location;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.utils.Pair;
+import com.wynntils.utils.StringUtils;
+import com.wynntils.wynn.model.CharacterManager;
+import com.wynntils.wynn.objects.ProfessionInfo;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TextComponent;
+
+public class QuestInfo {
+    private static final int NEXT_TASK_MAX_WIDTH = 200;
+    private static final Pattern COORDINATE_PATTERN = Pattern.compile(".*\\[(-?\\d+), ?(-?\\d+), ?(-?\\d+)\\].*");
+
+    private final String name;
+    private final QuestStatus status;
+    private final QuestLength length;
+    private final int level;
+    private String nextTask;
+    /** Additional requirements as pairs of <"profession name", minLevel> */
+    private final List<Pair<String, Integer>> additionalRequirements;
+
+    private final boolean isMiniQuest;
+    private final int pageNumber;
+    private final boolean tracked;
+
+    protected QuestInfo(
+            String name,
+            QuestStatus status,
+            QuestLength length,
+            int level,
+            String nextTask,
+            List<Pair<String, Integer>> additionalRequirements,
+            boolean isMiniQuest,
+            int pageNumber,
+            boolean tracked) {
+        this.name = name;
+        this.status = status;
+        this.length = length;
+        this.level = level;
+        this.nextTask = nextTask;
+        this.additionalRequirements = additionalRequirements;
+        this.isMiniQuest = isMiniQuest;
+        this.pageNumber = pageNumber;
+        this.tracked = tracked;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public QuestStatus getStatus() {
+        return status;
+    }
+
+    public boolean isTrackable() {
+        return status == QuestStatus.CAN_START || status == QuestStatus.STARTED;
+    }
+
+    public Optional<Location> getNextLocation() {
+        Matcher matcher = COORDINATE_PATTERN.matcher(ComponentUtils.stripFormatting(nextTask));
+        if (!matcher.matches()) return Optional.empty();
+
+        return Optional.of(new Location(
+                Integer.parseInt(matcher.group(1)),
+                Integer.parseInt(matcher.group(2)),
+                Integer.parseInt(matcher.group(3))));
+    }
+
+    public QuestLength getLength() {
+        return length;
+    }
+
+    public int getLevel() {
+        return level;
+    }
+
+    public String getNextTask() {
+        return nextTask;
+    }
+
+    public void setNextTask(String nextTask) {
+        this.nextTask = nextTask;
+    }
```

Comment:
I suggest renaming `task` to `step` as we use `task` already for render tasks

Reply:
And we also use "step" for the next step in queries, which is much tighter coupled with the QuestManager than rendering is. 

I've used the term `nextTask` consistently, so it's never called just `task`. 

I think we'll just have to accept that commonplace words like "step" and "task" can be used with different meaning in different contexts.